src/telemetry/telemetry.ts

```diff
@@ -110,3 +115,70 @@ export function sendErrorEvent(
     }
     errorEvent.send();
 }
+
+export function startActivity(
+    telemetryView: TelemetryViews,
+    telemetryAction: TelemetryActions,
+    correlationId?: string,
+    additionalProps: TelemetryEventProperties = {},
+    additionalMeasurements: TelemetryEventMeasures = {},
+): FinishActivityObject {
+    const startTime = performance.now();
+    if (!correlationId) {
+        correlationId = uuidv4();
+    }
+
+    sendActionEvent(telemetryView, telemetryAction, additionalProps, {
+        ...additionalMeasurements,
+        startTime: Math.round(startTime),
+    });
+
+    return {
+        correlationId: correlationId,
+        end: (
```

Comment:
IMO inline functions that span more than a couple lines are tough to read. Can you make this a local function and then reference that here?